docs/source/developers/guide/resources.rst

```diff
@@ -27,3 +27,41 @@
 ************************************
 Additional information and resources
 ************************************
+
+On this page we will list resources interesting for a contributor
+that is not yet acquainted with different parts of Apache Arrow.
+
+.. Annotation Glossary
+.. -------------------
+.. ARROW-15130
+
+Additional information
+----------------------
+
+- GitHub Actions
+
+  - `Apache Arrow Actions on GitHub <https://github.com/apache/arrow/actions>`_
+  - `GitHub Documentation on GitHub Actions <https://docs.github.com/en/actions>`_
```

Comment:
I'd be tempted to also add a link to the relevant directory in the codebase here too